Logic/igtlioConnector.cxx

```diff
@@ -701,6 +744,49 @@ CommandDevicePointer Connector::SendCommand(std::string device_id, std::string c
   return device;
 }
 
+//----------------------------------------------------------------------------
+void Connector::ParseCommands()
+{
+  this->CommandQueueMutex->Lock();
+  while (!this->CommandQueue.empty())
```

Comment:
If this done in a separate thread?

Reply:
Commands are added to the queue in the Receiver thread, and then parsed in the Main thread I believe.

Reply:
Hmm, although highly unlikely (which means don't change anything, just a discussion) this could theoretically block the main thread if an expensive command(s) came in. Reading a huge image volume from file, for example.

Reply:
Probably remove the while (!this->CommandQueue.empty()) loop? just get one command from the queue in every periodic process?

Reply:
@adamrankin Yeah, I see your point. I'm not sure that there's anything that can be done about it though. It probably needs to be dealt with by whatever application is executing the command.
The devices also need to be handled on the same thread that they're owned by, which tends to be the main thread (although maybe not always?).

Also, it was explained to me by @lassoan (correct me if I'm wrong Andras), that commands taking longer than a few seconds should send a quick acknowledgement response before sending a followup message later containing the requested info once it has been gathered.

@leochan2009 Sure.